driver/hinge_embedding_loss_driver.hpp

```diff
@@ -0,0 +1,701 @@
+    std::unique_ptr<GPUMem> output_dev;
+    std::unique_ptr<GPUMem> doutput_dev;
+    std::unique_ptr<GPUMem> dinput_dev;
+    std::unique_ptr<GPUMem> workspace_dev;
+
+    std::vector<TIO> input;
+    std::vector<TT> target;
+    std::vector<TIO> output;
+    std::vector<TIO> outputHost;
+    std::vector<TIO> doutput;
+    std::vector<TIO> dinput;
+    std::vector<TIO> dinputHost;
+    std::vector<TIO> workspace;
+
+    float margin;
+    float divisor;
+    bool isContiguous;
+    std::string reduction;
+
+    size_t workSpaceSizeInBytes;
+};
+
+template <typename TIO, typename TT>
+int HingeEmbeddingLossDriver<TIO, TT>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename TIO, typename TT>
+int HingeEmbeddingLossDriver<TIO, TT>::GetandSetData()
+{
+    std::vector<int> inDim = GetInputTensorLengthsFromCmdLine();
+    margin                 = inflags.GetValueDouble("margin");
+    isContiguous           = inflags.GetValueInt("is-contiguous") == 1 ? true : false;
+    reduction              = inflags.GetValueStr("reduction");
+
+    std::vector<int> inStride = GetTensorStride(inDim);
+    if(!isContiguous)
+    {
+        std::swap(inDim.front(), inDim.back());
+    }
+
+    SetTensorNd(inputDesc, inDim, inStride, data_type);
+    SetTensorNd(targetDesc, inDim, inStride, miopen_type<TT>{});
+    SetTensorNd(doutputDesc, inDim, data_type);
+    SetTensorNd(dinputDesc, inDim, data_type);
+
+    if(reduction == "none")
+    {
+        SetTensorNd(outputDesc, inDim, data_type);
+    }
+    else
+    {
+        std::vector<int> outDim(1);
+        outDim[0] = 1;
+        SetTensorNd(outputDesc, outDim, data_type);
+        divisor = 1;
+        if(reduction == "mean")
+        {
+            divisor = miopen::deref(inputDesc).GetElementSize();
+        }
+    }
+
+    return 0;
+}
+
+template <typename TIO, typename TT>
+std::vector<int> HingeEmbeddingLossDriver<TIO, TT>::GetTensorStride(std::vector<int> dim)
+{
+    std::vector<int> strides(dim.size(), 1);
+    for(int i = dim.size() - 2; i >= 0; --i)
+    {
+        strides[i] = dim[i + 1] * strides[i + 1];
+    }
+
+    if(!isContiguous)
+    {
+        std::swap(strides.front(), strides.back());
+    }
+
+    return strides;
+}
+
+template <typename TIO, typename TT>
+int HingeEmbeddingLossDriver<TIO, TT>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward (Default=1)", "int");
+    inflags.AddInputFlag("DimLengths",
+                         'D',
+                         "256,4,1,1,8723",
+                         "The dimensional lengths of the input tensor",
+                         "string");
+    inflags.AddInputFlag("is-contiguous", 'c', "1", "is-contiguous (Default=1)", "int");
+    inflags.AddInputFlag("reduction", 'R', "none", "reduction (Default='none')", "string");
```

Comment:
I think it would be more appropriate to receive this value as 1 and 0 rather than the string none.

Reply:
there are 3 options: none, sum, mean. I use string like pytorch uses for loss kernel. For ex: `torch.nn.HingeEmbeddingLoss(reduction='sum')`

Reply:
I have add explanation in driver and miopen.h for reduction flag/param.